docs/en/observability/cloud-monitoring/azure/monitor-azure-openai-apm.asciidoc

```diff
@@ -0,0 +1,167 @@
+[discrete]
+[[azure-openai-apm]]
+=== Step 7: Monitor Microsoft Azure OpenAI APM with OpenTelemetry
+
+The Azure OpenAI API provides useful information to help monitor and understand your code.
+To ingest this information into Elastic {observability}, use OpenTelemetry.
+
+For this tutorial, we'll be using an https://github.com/davidgeorgehope/ChatGPTMonitoringWithOtel[example application] and Python OpenTelemetry libraries.
+
+To start collecting APM data for your Azure OpenAI applications, you need to collect some information on your {cloud} instance:
+
+. From the {kib} homepage, click **Add integrations**.
+. Select the **APM** integration.
+. Make note of the configuration values from the following configuration settings:
+* `OTEL_EXPORTER_OTLP_ENDPOINT`
+* `OTEL_EXPORTER_OTLP_HEADERS`
+
+With the configuration values from the APM integration and your https://learn.microsoft.com/en-us/azure/ai-services/openai/quickstart?tabs=command-line%2Cpython-new&pivots=programming-language-python#retrieve-key-and-endpoint[Azure OpenAI API key], set the following environment values using these export commands on the command line:
+
+[source,bash]
+----
+export AZURE_OPENAI_API_KEY=<your-azure-openAI-key>
+export OTEL_EXPORTER_OTLP_AUTH_HEADER=<your-OTEL_EXPORTER_OTLP_AUTH-token>
+export OTEL_EXPORTER_OTLP_ENDPOINT=<your-OTEL_EXPORTER_OTLP_ENDPOINT>
+----
+
+Install the necessary Python libraries using the following commands:
+
+[source,bash]
+----
+pip3 install opentelemetry-api
+pip3 install opentelemetry-sdk
+pip3 install opentelemetry-exporter-otlp
+pip3 install opentelemetry-instrumentation
+pip3 install opentelemetry-instrumentation-requests
+pip3 install openai
```

Comment:
This is the openai library. Shouldn't we be using the azure openai library? I don't know if such library exist for python (it does for javascript).

Reply:
Thanks for the tip, I found this [page](https://learn.microsoft.com/en-us/azure/ai-services/openai/how-to/switching-endpoints) so I'll give this a shot.

Reply:
It turned out that we needed to import AzureOpenAI from openai, and the `Completion` call was deprecated and needed to be replaced with `chat.completions`, which also required some additional changes in formatting. But all seems to be working now. Thinks for pointing me in the right direction.